Microsoft.TeamServices.Samples.Client/Release/ReleasesSample.cs

```diff
@@ -0,0 +1,512 @@
+            int releaseDefinitionId = releaseDefinitions.FirstOrDefault().Id;
+            ReleaseDefinition releaseDefinition = releaseClient.GetReleaseDefinitionAsync(project: projectName, definitionId: releaseDefinitionId).Result;
+
+            // add a non secret variable to definition
+            ConfigurationVariableValue variable = new ConfigurationVariableValue();
+            variable.Value = "NonSecretValue";
+            variable.IsSecret = false;
+            releaseDefinition.Variables.Add("NonSecretVariable", variable);
+
+            // update release definition
+            ReleaseDefinition updatedReleaseDefinition = releaseClient.UpdateReleaseDefinitionAsync(project: projectName, releaseDefinition: releaseDefinition).Result;
+
+            Console.WriteLine("{0} {1}", updatedReleaseDefinition.Id.ToString().PadLeft(6), updatedReleaseDefinition.Name);
+
+            return releaseDefinition;
+        }
+
+        [ClientSampleMethod]
+
+        public IEnumerable<ReleaseDefinitionRevision> GetReleaseDefinitionRevisions()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient releaseClient = connection.GetClient<ReleaseHttpClient>();
+
+            List<ReleaseDefinition> releaseDefinitions = releaseClient.GetReleaseDefinitionsAsync(project: projectName, searchText: releaseDefinitionName).Result;
+            int releaseDefinitionId = releaseDefinitions.FirstOrDefault().Id;
+
+            // Get revision
+            List<ReleaseDefinitionRevision> revisions = releaseClient.GetReleaseDefinitionHistoryAsync(projectName, releaseDefinitionId).Result;
+
+            foreach (ReleaseDefinitionRevision revision in revisions)
+            {
+                Console.WriteLine("{0} {1}", revision.DefinitionId.ToString().PadLeft(6), revision.Revision);
+            }
+
+            return revisions;
+        }
+
+        [ClientSampleMethod]
+
+        public System.IO.Stream GetReleaseDefinitionForARevision()
+        {
+            string projectName = ClientSampleHelpers.FindAnyProject(this.Context).Name;
+
+            // Get a release client instance
+            VssConnection connection = Context.Connection;
+            ReleaseHttpClient releaseClient = connection.GetClient<ReleaseHttpClient>();
+
+            List<ReleaseDefinition> releaseDefinitions = releaseClient.GetReleaseDefinitionsAsync(project: projectName, searchText: releaseDefinitionName).Result;
+            int releaseDefinitionId = releaseDefinitions.FirstOrDefault().Id;
+
+            // Get revision
+            System.IO.Stream revision = releaseClient.GetReleaseDefinitionRevisionAsync(projectName, releaseDefinitionId, revision: 1).Result;
```

Comment:
returning Stream were byDesign as we get the data from fileservice. But i shall confirm it once. What issue do you see with stream?